lib/driver.rb

```diff
@@ -0,0 +1,75 @@
+require_relative 'csv_record'
+
+module RideShare
+  class Driver < CsvRecord
+    attr_reader :id, :name, :vin, :status, :trips
+
+    def initialize(id:, name:, vin:, status: :AVAILABLE, trips: nil)
+
+      super(id)
+      @name = name
+      @vin = vin
+      raise ArgumentError.new("Incorrect VIN length.") if vin.length != 17
+
+      @status = status
+      raise ArgumentError.new("Invalid status.") unless [:AVAILABLE, :UNAVAILABLE].include? status
+
+      @trips = trips || []
+    end
+
+    def add_trip(trip)
+      @trips << trip
+    end
+
+    def average_rating
+      return 0 if trips.length == 0
+
+      finished_trips = 0
+      rating_sum = 0
+      trips.each do |trip|
+        if trip.rating != nil
+          finished_trips += 1
+          rating_sum += trip.rating
+        end
```

Comment:
You can also use some enumerables as well to simplify this a bit.

```suggestion
      complete_trips = trips.select { |trip| !trip.nil? }
      total = complete_trips.reduce(0) do |sum, trip|
        sum + trip.cost
      end
      
      return total / complete_trips.length.to_f
```